Tasks/Common/ios-signing-common/ios-signing-common.ts

```diff
@@ -602,8 +598,83 @@ export async function getP12CommonName(p12Path: string, p12Pwd: string) {
     return commonName;
 }
 
-export async function deleteCert(keychainPath: string, certSha1Hash: string) {
+export async function deleteCert(keychainPath: string, certSha1Hash: string): Promise<void> {
     let deleteCert: ToolRunner = tl.tool(tl.which('security', true));
     deleteCert.arg(['delete-certificate', '-Z', certSha1Hash, keychainPath]);
     await deleteCert.exec();
 }
+
+export async function getP12PrivateKeyName(p12Path: string, p12Pwd: string): Promise<string> {
+    //openssl pkcs12 -in <p12Path> -nocerts -passin pass:"<p12Pwd>" -passout pass:"<p12Pwd>" | grep 'friendlyName'
+    //We pipe through grep so we we don't log the private key, even encrypted.
+    tl.debug('getting the P12 private key name');
+    let opensslPath: string = tl.which('openssl', true);
+    let openssl: ToolRunner = tl.tool(opensslPath);
+    if (!p12Pwd) {
+        // if password is null or not defined, set it to empty
+        p12Pwd = '';
+    }
+    openssl.arg(['pkcs12', '-in', p12Path, '-nocerts', '-passin', 'pass:' + p12Pwd, '-passout', 'pass:' + p12Pwd]);
+
+    let grepPath: string = tl.which('grep', true);
+    let grep: ToolRunner = tl.tool(grepPath);
```

Comment:
Is this being used?  It looks like the `on stdline` event handler is doing the same thing with a regex

Reply:
Yes, by piping through grep to find the friendlyName line, all of this doesn't get logged to the console:

```
Bag Attributes
    friendlyName: Mac Developer: Chris Sidi (Chris Sidi)
    localKeyID: A0 A0 EB 6F 7A F0 3E F7 26 A1 5A 8A 7B 82 E0 6B A5 F8 63 5D 
Key Attributes: <No Attributes>
-----BEGIN ENCRYPTED PRIVATE KEY-----
MIIFDjBABgkqhkiG9w0BBQ0wMzAbBgkqhkiG9w0BBQwwDgQIzNkMhsUKHvUCAggA
MBQGCCqGSIb3DQMHBAjfDeUtX5Vc7ASCBMg39GmfcF9edu1y6eO1Useu9er5XH/k
r3VbGyJMVRYRu64tQZmw2mLO+oXdy6JbEj5xcEQqMThgWMzYV+U/RAQS/T7f0SXc
mk3vB7Ke9qGfvKT1UdDjQiM1H038onh3MdL/ArhNTxAJLRYuhFogjs25saZ8ucMM
B6QUkCPrVMc08N0a9NxYwCTEMVMLmh8vcPqJkWRmgfRjuTnMDgwIHRiPh8LB1s68
Mn6OvSAPXmrQdqK65R7ioHYuLOYF0Ck15dUJqZsBzXZxK3jryDyq340vgk7BJW+r
...
```

Reply:
Ah ok, that's smart.  Maybe we should just add a comment here to say "filter console output."  It's clear if you know the details of how tool runner works, but I feel like it's just a little subtle.